pyramid/config/views.py

```diff
@@ -762,6 +782,17 @@ def view(context, request):
         if context is None:
             context = for_
 
+        if exception is not None:
+            if context is not None:
+                raise ConfigurationError('view "context" and "exception" '
+                                         'arguments are mutually exclusive')
+            context = exception
+
+        isexc = isexception(context)
+        if not isexc and exception is not None:
+            raise ConfigurationError(
+                'view "exception" must be an exception type')
```

Comment:
I think making that public API would probably be a good idea so that people could use that in various other parts of their code as needed.